docs/reference/connectors.md

```diff
@@ -0,0 +1,90 @@
+# Connectors
+
+This resource packs supports the following connection configurations:
+
+* Environment Variables
+* Credentials file
+* Managed Service Identity Connector
+
+For any of these connection strategies you'll need a subscription id. Your subscription id may be found by ... When you run InSpec it will be included in the target. `-t azure://$SUBSCRIPTION_ID`.
+
+## Environment Variables
+
+Your account information will be read off of environment variables.
+
+On Linux:
+```
+export AZURE_CLIENT_ID=<client id>
+export AZURE_TENANT_ID=<tenant id>
+export AZURE_CLIENT_SECRET=<client secret>
+```
+
+On Windows:
+``
+$env:AZURE_CLIENT_ID="<client id>"
+$env:AZURE_CLIENT_SECRET="<tenant id>"
+$env:AZURE_TENANT_ID="<client secret>"
+```
+## Credentials File
+
+A credentials file may also be used. The default location is
+`~/.azure/credentials`. You may override the default location by setting the
+environment varialbe `AZURE_CRED_FILE` with the path you wish to use. The
+credentials file is a ini file with the following structure:
+
+```
+[SUBSCRIPTION_ID]
+TENANT_ID = <tenant id>
+CLIENT_ID = <client id>
+CLIENT_SECRET = <client secret>
+```
+
+Multiple subscription ids may be added by adding additional `[SUBSCRIPTION_ID]` sections:
+
+```
+[SUBSCRIPTION_ID_1]
+TENANT_ID = <tenant id 1>
+CLIENT_ID = <client id 1>
+CLIENT_SECRET = <client secret 1>
+
+[SUBSCRIPTION_ID_2]
+TENANT_ID = <tenant id 2>
+CLIENT_ID = <client id 2>
+CLIENT_SECRET = <client secret 2>
+```
+
+The Azure train connector will select the correct information based on the
+subscription id you pass in the target parameter: `-t
+azure://SUBSCRIPTION_ID_2` would choose the second entry in the credentials
+file.
+
+If you have a single entry in your credentials file, then you may omit the
+subscription id from the target flag.
+
+`-t azure://`
+
+You may also set an environment variable `AZURE_SUBSCRIPTION_NUMBER` to chose a
+subscription based on its index in the file. The index is 1 based with this
+environment variable. So if you wish to select the second entry in a file with
+two entries:
+
+```
+$ export AZURE_SUBSCRIPTION_NUMBER=2
+$ bundle exec insepc exec . -t azure://
+```
+
+## Managed Service Identity Connector
+
+A Managed Service Identity (MSI) connector may be used as well. For more
+information about MSI see
+[What is Managed Service Identity for Azure resources?](https://docs.microsoft.com/en-us/azure/active-directory/managed-service-identity/overview).
+Using this strategy will require that you have enabled MSI on the virtual
+machine you wish to use.
+
+Using this method you will only need to have a subscription id, which should be
+included with the target flag `-t azure://SUBSCRIPTION_ID`. When we detect that
```

Comment:
Reword: We will use MSI when there is no Client ID and no Client Secret and the MSI port is open (50342 by default). You may change the default port by setting an environment variable `AZURE_MSI_PORT` with the port you have configured.